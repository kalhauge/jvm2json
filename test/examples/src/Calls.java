public class Calls {

    public int first(int a, float b) {
        return a;
    }

    public void main(String [] args) {
        System.out.println(new Calls().first(10, 20.0f));
    }

}
