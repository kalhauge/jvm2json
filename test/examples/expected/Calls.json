{
  "name": "Calls",
  "access": [
    "public",
    "super"
  ],
  "typeparams": [],
  "super": {
    "name": "java/lang/Object",
    "inner": null,
    "args": [],
    "annotations": []
  },
  "interfaces": [],
  "fields": [],
  "methods": [
    {
      "name": "<init>",
      "access": [
        "public"
      ],
      "typeparams": [],
      "params": [],
      "returns": {
        "type": null,
        "annotations": []
      },
      "code": {
        "max_stack": 1,
        "max_locals": 1,
        "exceptions": [],
        "stack_map": null,
        "bytecode": [
          {
            "offset": 0,
            "opr": "load",
            "type": "ref",
            "index": 0
          },
          {
            "offset": 1,
            "opr": "invoke",
            "access": "special",
            "method": {
              "is_interface": false,
              "ref": {
                "kind": "class",
                "name": "java/lang/Object"
              },
              "name": "<init>",
              "args": [],
              "returns": null
            }
          },
          {
            "offset": 4,
            "opr": "return",
            "type": null
          }
        ]
      },
      "annotations": [],
      "exceptions": [],
      "default": null
    },
    {
      "name": "first",
      "access": [
        "public"
      ],
      "typeparams": [],
      "params": [
        {
          "visible": true,
          "type": {
            "base": "int",
            "annotations": []
          },
          "annotations": []
        },
        {
          "visible": true,
          "type": {
            "base": "float",
            "annotations": []
          },
          "annotations": []
        }
      ],
      "returns": {
        "type": {
          "base": "int"
        },
        "annotations": []
      },
      "code": {
        "max_stack": 1,
        "max_locals": 3,
        "exceptions": [],
        "stack_map": null,
        "bytecode": [
          {
            "offset": 0,
            "opr": "load",
            "type": "int",
            "index": 1
          },
          {
            "offset": 1,
            "opr": "return",
            "type": "int"
          }
        ]
      },
      "annotations": [],
      "exceptions": [],
      "default": null
    },
    {
      "name": "main",
      "access": [
        "public"
      ],
      "typeparams": [],
      "params": [
        {
          "visible": true,
          "type": {
            "kind": "array",
            "type": {
              "kind": "class",
              "name": "java/lang/String",
              "inner": null,
              "args": [],
              "annotations": []
            },
            "annotations": []
          },
          "annotations": []
        }
      ],
      "returns": {
        "type": null,
        "annotations": []
      },
      "code": {
        "max_stack": 4,
        "max_locals": 2,
        "exceptions": [],
        "stack_map": null,
        "bytecode": [
          {
            "offset": 0,
            "opr": "get",
            "static": true,
            "field": {
              "class": "java/lang/System",
              "name": "out",
              "type": {
                "kind": "class",
                "name": "java/io/PrintStream"
              }
            }
          },
          {
            "offset": 3,
            "opr": "new",
            "class": "Calls"
          },
          {
            "offset": 6,
            "opr": "dup",
            "words": 1
          },
          {
            "offset": 7,
            "opr": "invoke",
            "access": "special",
            "method": {
              "is_interface": false,
              "ref": {
                "kind": "class",
                "name": "Calls"
              },
              "name": "<init>",
              "args": [],
              "returns": null
            }
          },
          {
            "offset": 10,
            "opr": "push",
            "value": {
              "type": "integer",
              "value": 10
            }
          },
          {
            "offset": 12,
            "opr": "push",
            "value": {
              "type": "float",
              "value": 20
            }
          },
          {
            "offset": 14,
            "opr": "invoke",
            "access": "virtual",
            "method": {
              "ref": {
                "kind": "class",
                "name": "Calls"
              },
              "name": "first",
              "args": [
                "int",
                "float"
              ],
              "returns": "int"
            }
          },
          {
            "offset": 17,
            "opr": "invoke",
            "access": "virtual",
            "method": {
              "ref": {
                "kind": "class",
                "name": "java/io/PrintStream"
              },
              "name": "println",
              "args": [
                "int"
              ],
              "returns": null
            }
          },
          {
            "offset": 20,
            "opr": "return",
            "type": null
          }
        ]
      },
      "annotations": [],
      "exceptions": [],
      "default": null
    }
  ],
  "bootstrapmethods": [],
  "enclosingmethod": null,
  "innerclasses": [],
  "annotations": [],
  "version": [
    63,
    0
  ]
}
